pageserver/src/tenant/timeline.rs

```diff
@@ -1906,17 +1909,16 @@ impl Timeline {
             // need to return something
             Ok(0)
         });
-        let timer = if up_to_lsn == self.initdb_lsn {
-            if let Some(size) = self.current_logical_size.initialized_size() {
-                if size != 0 {
-                    // non-zero size means that the size has already been calculated by this method
-                    // after startup. if the logical size is for a new timeline without layers the
-                    // size will be zero, and we cannot use that, or this caching strategy until
-                    // pageserver restart.
-                    return Ok(size);
-                }
+        // See if we've already done the work for initial size calculation.
+        // This is a short-cut for timelines that are mostly unused.
+        let timer = if let Some(size) = self.current_logical_size.initialized_size(up_to_lsn) {
+            if size != 0 {
+                // non-zero size means that the size has already been calculated by this method
+                // after startup. if the logical size is for a new timeline without layers the
+                // size will be zero, and we cannot use that, or this caching strategy until
+                // pageserver restart.
+                return Ok(size);
```

Comment:
Would be nice to understand
1. why 0 needs special treatment. I don't get it. Must be something about branches.
2. whether we can put this logic into `fn initialized_size()`

Reply:
You could try to remove the `if size != 0` check, and run the tests and see what happens.

Reply:
Talked this over with @koivunej , we can remove it.